src/components/App/App.js

```diff
@@ -29,25 +29,36 @@ class App extends Component {
 
   displayAllMovies = () => {
     Promise.resolve(fetchData('movies'))
-        .then(data => {
-          data.movies.sort((a, b) => {
-            if (a.title > b.title) {
-              return 1 
-            } else {
-              return -1
-            }
-          })
-          this.setState({movies: data.movies, movieData: {}, showDetails: false})
-        })
-        .catch(error => {
-          console.log(error)
-          this.setState({error: 'Something went wrong, please try again later.'})
-        })
+      .then(data => {
+        this.sortByTitle(data.movies)
+      })
+      .catch(error => {
+        console.log(error)
+        this.setState({movies: [], error: 'Something went wrong, please try again later.'})
+      })
+  }
+
+  sortByRating = (data) => {
+    data.sort((a, b) => {
+      return b.rating - a.rating;
+    })
+    this.setState({movies: data, movieDetails: {}, showDetails: false})
+  }
+
+  sortByTitle = (data) => {
+    data.sort((a, b) => {
+      if (a.title > b.title) {
+        return 1 
+      } else {
+        return -1
+      }
+    })
+    this.setState({movies: data, movieDetails: {}, showDetails: false})
   }
 
   render() {
     const allMovieData = this.state.movies.map(movie => {
-      return {key: movie.id, posterPath: movie['poster_path'], title: movie.title, rating: movie['average_rating']}
+      return {key: movie.id || movie.key, posterPath: movie['poster_path'] || movie.posterPath, title: movie.title, rating: movie['average_rating'] || movie.rating}
```

Comment:
We talked about this in person but we should clarify if this is what the rubric was referring to  when they mentioned "cleaning the code in a utility file". Here it seems like we are cleaning as we go when we could set start with clean code and the key names from the start and stick to them throughout our prop usage.

Reply:
But lets maybe clarify that this week and then refactor when we have more insight. Looks awesome otherwise!!